crates/primitives/src/utils/mod.rs

```diff
@@ -75,10 +78,10 @@ pub fn box_try_new_uninit<T>() -> Result<Box<MaybeUninit<T>>, TryReserveError> {
     // Make sure we got exactly 1 element.
     vec.shrink_to(1);
 
-    let ptr = vec.as_mut_ptr();
-    core::mem::forget(vec);
+    let mut vec = ManuallyDrop::new(vec);
+
     // SAFETY: `vec` is exactly one element long and has not been deallocated.
-    Ok(unsafe { Box::from_raw(ptr) })
+    Ok(unsafe { Box::from_raw(vec.as_mut_ptr()) })
```

Comment:
`Vec<T>` of length 1 capacity 1 is the exact same as a `Box<T>`

Reply:
Okay this is safe because Align of array = Align of elem and because vec is allocated for exactly 1 elem. So their layouts are equal. But this is shady ><'

Reply:
Why don't you alloc for T but vec<T> with 1 elem?

Reply:
somewhat of a hack for box::try_new as it's unstable